lib/puppet/util/windows/eventlog.rb

```diff
@@ -115,6 +116,63 @@ def initialize(msg, code)
     end
   end
 
+  def create_custom_event_log(name)
+    raise ArgumentError, "Invalid custom log name" if name.nil?
+
+    require 'win32/registry'
+    include Win32::Registry::Constants
+
+    begin
+      Win32::Registry::HKEY_LOCAL_MACHINE.create('SYSTEM\CurrentControlSet\Services\Eventlog\Puppet', KEY_ALL_ACCESS | 0x0100) do |reg_parent|
+        # // The reasoning behind filling these values is historical. WS03 RTM had a ----
```

Comment:
This code comes from running DotPeek over the PowerShell source and as such can't be included like this.